### PR TITLE
Multi distro travis

### PR DESCRIPTION
**Pull request by fmessmer**
replaces #620 
includes #621 

no kinetic_dev needed

### CODE REVIEW

**Comment by fmessmer** [comment]
> The job exceeded the maximum time limit for jobs, and has been terminated.

...oh boy :man_facepalming:

**Comment by mathias-luedtke** [comment]
I am preparing a patch for industrial_ci that should speed up the build by filtering unrelated packges.

**Comment by mathias-luedtke** [comment]
You can use another docker image to speed thinks up:
```
DOCKER_IMAGE=ros:kinetic-robot  ADDITIONAL_DEBS="build-essential python-catkin-tools python-pip"
```
Beware, this prevents detecting missing dependency tags.

**Comment by mathias-luedtke** [comment]
I have just noticed the build was finished just in time.
For now you can just set `NOT_TEST_INSTALL=true`. This prevents rebuilding everything from scratch for install space.

**Comment by fmessmer** [comment]
rebased on top of #621

**Comment by fmessmer** [comment]
as discussed, we should not use option 2 (docker image) or option 3 (NOT_TEST_INSTALL=true)...
if @ipa-mdl's filter for industrial_ci works...fine...
still, releasing an update to indigo/kinetic will be the best solution...as it would clear the .rosinstall files.

other options, we might want to critically review dependencies and try to find stuff that is not necessarily needed...

other than that, @ipa-fmw, don't we pay for travis? shouldn't that grant us some advantages, i.e. sufficient build time?

**Comment by mathias-luedtke** [comment]
Without limiting the tests:
* wait for my industial_ci PR which handles dependencies better
* downstrip external dependencies

**Comment by floweisshardt** [comment]
see https://docs.travis-ci.com/user/customizing-the-build/#Build-Timeouts for Travis build timeouts

**Comment by mathias-luedtke** [comment]
> wait for my industial_ci PR which handles dependencies better

here it is: https://github.com/ros-industrial/industrial_ci/pull/137, it would be great if you could test it

**Comment by fmessmer** [comment]
Tested @ipa-mdl's enhancement from https://github.com/ros-industrial/industrial_ci/pull/137 locally on my laptop in `cob_robots` (not the meta package :stuck_out_tongue:)  by running:
```
rosrun industrial_ci run_ci ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu UPSTREAM_WORKSPACE=file
```
(Well, I again had to add `python` explicitly to the call of deps.py)

```
ici_fold:end:setup_workspace<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Function setup_workspace returned with code '0' after 2 min 54 sec 

Upstream packages: canopen_402 canopen_chain_node canopen_master canopen_motor_node cob_android_msgs cob_android_script_server cob_base_drive_chain cob_base_velocity_smoother cob_bms_driver cob_calibration_data cob_cam3d_throttle cob_canopen_motor cob_collision_monitor cob_collision_velocity_filter cob_command_gui cob_control_mode_adapter cob_control_msgs cob_dashboard cob_default_env_config cob_description cob_docker_control cob_footprint_observer cob_frame_tracker cob_generic_can cob_hand_bridge cob_helper_tools cob_image_flip cob_light cob_linear_nav cob_mimic cob_monitoring cob_msgs cob_obstacle_distance cob_omni_drive_controller cob_perception_msgs cob_phidget_em_state cob_phidget_power_state cob_phidgets cob_reflector_referencing cob_relayboard cob_safety_controller cob_scan_unifier cob_script_server cob_sick_lms1xx cob_sick_s300 cob_sound cob_srvs cob_supported_robots cob_teleop cob_trajectory_controller cob_twist_controller cob_undercarriage_ctrl cob_utilities cob_voltage_control libntcan libpcan libphidgets raw_description schunk_description schunk_libm5api schunk_powercube_chain sick_visionary_t_driver socketcan_interface ur_description ur_driver ur_msgs usb_cam

Target packages: cob_bringup cob_default_robot_behavior cob_default_robot_config cob_hardware_config cob_moveit_config cob_robots
Target tests: cob_bringup cob_default_robot_behavior cob_default_robot_config cob_hardware_config cob_moveit_config cob_robots

Downstream packages: 
Downstream tests: 

Ignored packages: can_msgs cob_3d_mapping_msgs cob_android cob_android_resource_server cob_android_settings cob_camera_sensors cob_cartesian_controller cob_command_tools cob_common cob_control cob_driver cob_elmo_homing cob_environments cob_extern cob_grasp_generation cob_hand cob_head_axis cob_interactive_teleop cob_lookat_action cob_manipulation cob_map_accessibility_analysis cob_mapping_slam cob_model_identifier cob_moveit_bringup cob_moveit_interface cob_navigation cob_navigation_config cob_navigation_global cob_navigation_local cob_navigation_slam cob_object_detection_msgs cob_object_detection_visualizer cob_obstacle_distance_moveit cob_perception_common cob_pick_place_action cob_substitute cob_undercarriage_ctrl_node cob_vision_utils libconcorde_tsp_solver libdlib libqsopt opengm ros_canopen schunk_modular_robotics schunk_sdh schunk_sdhx schunk_simulated_tactile_sensors service_tools socketcan_bridge universal_robot ur10_moveit_config ur3_moveit_config ur5_moveit_config ur_bringup ur_gazebo ur_kinematics
```

Unfortunately, I first ran out of memory during `rosdep install`....I definitely will provide https://github.com/ipa320/cob_control/pull/145/commits/8453f1aa76acea939ad23e2927e4aee539438652 in a separate PR....then, I cleaned up some old stuff and now I am running it again....

...then it started to compile and killed my laptop....this is to much for it :wink: 


---

In addition to speeding up Travis CI, it is a great tool for finding missed dependencies :wink: 
e.g. `cob_cartesian_controller` should definitely be an `<exec_depend>`....need to go through the others, too

It might also help to identify "dead" packages...e.g. `cob_camera_sensors`

**Comment by mathias-luedtke** [comment]
The current implementation assumes that we link all test dependencies.
However, `cob_bringup` does not reference it's dependencies for the roslaunch checks.

**Comment by fmessmer** [comment]
restarting travis after https://github.com/ipa320/cob_control/pull/147 was merged...

File: .travis.rosinstall.kinetic [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    version: indigo_dev
+- git:
+    uri: 'https://github.com/ipa320/cob_supported_robots.git'
+    local-name: cob_supported_robots
+    version: indigo_dev
+- git:
+    uri: 'https://github.com/ipa320/schunk_modular_robotics.git'
+    local-name: schunk_modular_robotics
+    version: indigo_dev
+
```
Comment (NadiaHG) [inline .travis.rosinstall.kinetic:61]: please remove empty lines

**Review by NadiaHG — COMMENTED** [review]
~~missed .travis.rosinstall file, cob_robots should be always tested with the current indigo_dev branches, at least for cob_calibration_data and cob_supported_robots~~ it is already there https://github.com/ipa320/cob_robots/blob/indigo_dev/.travis.rosinstall

File: .travis.rosinstall.kinetic (hunk 1 above)
Comment (fmessmer) [reply .travis.rosinstall.kinetic:61]: I used the blank line to separate repositories which we want to use from release but currently are not released for kinetic yet...it's kind of "the foreign" packages...i.e. not `cob_` packages

**Comment by fmessmer** [comment]
> missed .travis.rosinstall file

`.travis.rosinstall` (for indigo) is already availablie in `indigo_dev` (nothings needs to be changed)

 > cob_robots should be always tested with the current indigo_dev branches, at least for cob_calibration_data and cob_supported_robots

this is what @ipa-mdl commented this morning, i.e. it's a "downstream" test...
I will create a new issue for it https://github.com/ipa320/care-o-bot/issues/39

**Review by NadiaHG — APPROVED** [review]

**Comment by fmessmer** [comment]
Travis for indigo succeeded
Travis for kinetic failed due to timeout...will merge anyway...as this will be fixed after first `cob_*` packages have been released for kinetic and thus, we can remove some entries from the .rosinstall.kinetic file

**Pull request merged**